modules/light-clients/08-wasm/types/misbehaviour_handle_test.go

```diff
@@ -1,499 +1,92 @@
 package types_test
 
-/* func (suite *TypesTestSuite) TestVerifyMisbehaviourGrandpa() {
-	var (
-		ok          bool
-		clientMsg   exported.ClientMessage
-		clientState exported.ClientState
-	)
+import (
+	"encoding/json"
+	fmt "fmt"
+
+	wasmvm "github.com/CosmWasm/wasmvm"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+
+	errorsmod "cosmossdk.io/errors"
+
+	wasmtesting "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/testing"
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+	ibcerrors "github.com/cosmos/ibc-go/v8/modules/core/errors"
+	"github.com/cosmos/ibc-go/v8/modules/core/exported"
+	ibctmtypes "github.com/cosmos/ibc-go/v8/modules/light-clients/07-tendermint"
+)
+
+func (suite *TypesTestSuite) TestVerifyClientMessage() {
+	var clientMsg exported.ClientMessage
 
 	testCases := []struct {
 		name     string
 		malleate func()
-		expPass  bool
+		expErr   error
 	}{
 		{
-			"successful misbehaviour verification",
+			"success: valid misbehaviour",
 			func() {
-				data, err := base64.StdEncoding.DecodeString(suite.testData["header"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
-				// VerifyClientMessage must be run first
-				err = clientState.VerifyClientMessage(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
-				suite.Require().NoError(err)
-				clientState.UpdateState(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
+				suite.mockVM.RegisterQueryCallback(types.VerifyClientMessageMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) ([]byte, uint64, error) {
```

Comment:
same recommendation as other PR, we should unmarshal `queryMsg []byte` into the expected type and make some assertions in here